baselines/fedpara/fedpara/dataset.py

```diff
@@ -0,0 +1,125 @@
+                ]
+            )
+            try:
+                with open(ds_path, "rb") as file:
+                    train_datasets = pickle.load(file)
+                dataset_train = Dataset(
+                    data_directory,
+                    train=True,
+                    download=False,
+                    transform=transform_train,
+                )
+                dataset_test = Dataset(
+                    data_directory,
+                    train=False,
+                    download=False,
+                    transform=transform_test,
+                )
+            except FileNotFoundError:
+                dataset_train = Dataset(
+                    data_directory, train=True, download=True, transform=transform_train
+                )
+                if config.partition == "iid":
+                    train_datasets = iid(dataset_train, num_clients)
+                else:
+                    train_datasets, _ = noniid(dataset_train, num_clients, config.alpha)
+                pickle.dump(train_datasets, open(ds_path, "wb"))
+                train_datasets = train_datasets.values()
+                dataset_test = Dataset(
+                    data_directory, train=False, download=True, transform=transform_test
+                )
+
+        case "MNIST":
+            ds_path = f"{data_directory}train_{num_clients}.pkl"
+            transform_train = transforms.Compose(
+                [
+                    transforms.ToTensor(),
+                    transforms.Normalize((0.1307,), (0.3081,)),
+                ]
+            )
+            transform_test = transforms.Compose(
+                [transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))]
+            )
+            try:
+                train_datasets = pickle.load(open(ds_path, "rb"))
+                dataset_train = Dataset(
+                    data_directory,
+                    train=True,
+                    download=False,
+                    transform=transform_train,
+                )
+                dataset_test = Dataset(
+                    data_directory,
+                    train=False,
+                    download=False,
+                    transform=transform_test,
+                )
+            except FileNotFoundError:
+                dataset_train = Dataset(
+                    data_directory, train=True, download=True, transform=transform_train
+                )
+                train_datasets = mnist_niid(
+                    dataset_train, num_clients, config.shard_size, config.data_seed
+                )
+                pickle.dump(train_datasets, open(ds_path, "wb"))
+                dataset_test = Dataset(
+                    data_directory, train=False, download=True, transform=transform_test
+                )
+
+    test_loader = DataLoader(dataset_test, batch_size=batch_size, num_workers=2)
+    train_loaders = [
+        DataLoader(
+            DatasetSplit(dataset_train, ids),
```

Comment:
There seems too be a problem with `ids`. It is an integer but inside `DatasetSplit` you do `list()`. That raises an error....

Reply:
Should it be replaced instead with `[ ]`?